libgnucash/engine/test/utest-Budget.c

```diff
@@ -96,10 +96,10 @@ test_gnc_set_budget_num_periods_data_retention ()
     gnc_budget_set_num_periods(budget, 10);
     gnc_budget_set_num_periods(budget, 20);
 
-    /* value and note are retained */
-    g_assert (gnc_budget_is_account_period_value_set(budget, acc, 15));
+    /* value and note are lost */
+    g_assert (!gnc_budget_is_account_period_value_set(budget, acc, 15));
     note = gnc_budget_get_account_period_note (budget, acc, 11);
-    g_assert_cmpstr (note, ==, "undefined");
+    g_assert_cmpstr (note, ==, NULL);
```

Comment:
I'd think this change of behaviour isn't considered unacceptable...

Reply:
I don't really understand the point. Asserts are for detecting programming errors when running unit tests and aren't much use for errors in the wild.

Reply:
Oh, this is a unit test. Never mind.